promql/lex.go

```diff
@@ -571,7 +571,7 @@ func lexValueSequence(l *lexer) stateFn {
 		l.emit(itemBlank)
 	case unicode.IsDigit(r) || (r == '.' && unicode.IsDigit(l.peek())):
 		l.backup()
-		lexNumber(l)
+		lexNumberOrDuration(l)
```

Comment:
Let's do this change separately instead of including it here without first resolving the discussion in  https://github.com/prometheus/prometheus/issues/419.